torch/cuda/memory.py

```diff
@@ -738,7 +738,7 @@ def _record_memory_history_legacy(
     )
 
 
-def _record_memory_history(enabled="all", *args, **kwargs):
+def _record_memory_history(enabled: Literal[None, "state", "all"] = "all", *args, **kwargs) -> None:
```

Comment:
```suggestion
def _record_memory_history(enabled: Optional[Literal["state", "all"]] = "all", *args, **kwargs) -> None:
```
minor nit: although the original is valid too. Fine keeping if it causes a nasty line wraparound.

Reply:
I just set it to match the docstring. Would you leave it as is or change it to `Optional`? And if we change it to `Optional` would you like me to update the docstring as well?